lib/Util.php

```diff
@@ -10,76 +10,89 @@
 class Util {
 
 	/**
-	 * Wrapper for print_r for quick debugging.
-	 * Debugging should be done with Xdebug.
+	 * Remove the query string from a url.
 	 *
-	 * @param string $label The button label.
-	 * @param array  $var   The var to display.
+	 * @param string $url The url to clean.
 	 */
-	public static function print( $label, $var ) : void {
-		// phpcs:disable WordPress.PHP.DevelopmentFunctions.error_log_print_r
-		?>
-		<br class="clear" />
-		<details>
-			<summary class="button button-secondary button-large" style="user-select: none; opacity: 0.5;"><?php echo esc_html( $label ); ?></summary>
-			<pre style="background-color: #333; color: #fff; padding: 10px; margin-right: 20px; border-radius: 3px;"><?php print_r( $var ); ?></pre>
-		</details>
-		<?php
-		// phpcs:enable WordPress.PHP.DevelopmentFunctions.error_log_print_r
+	public static function remove_query_string( $url ): string {
+		return preg_replace( '/\?.*/', '', $url );
 	}
 
 	/**
-	 * Check if string for substring.
-	 * Improves readability.
+	 * Sanitize our image URLs:
+	 *
+	 * Update .tif images to be .png.
+	 * Remove query strings.
 	 *
-	 * @param string $needle   The substring.
-	 * @param string $haystack The string to search.
+	 * @param string $image_url The image URL to sanitize.
 	 */
-	public static function contains( $needle, $haystack ) : bool {
-		return strpos( $haystack, $needle ) !== false;
+	public static function sanitize_image_url( $image_url ): string {
+		// Update .tif images to be .png.
+		if ( strpos( $image_url, '.tif' ) !== false ) {
+			$image_url = str_replace( '.tif', '.png', $image_url );
+		}
+
+		// Remove query string.
+		$image_url = self::remove_query_string( $image_url );
+
+		return $image_url;
 	}
 
 	/**
-	 * Remove the query string from a url.
+	 * Wrapper for error_log that handles both arrays and strings.
 	 *
-	 * @param string $url The url to clean.
+	 * @param string|array $log What we want logged.
+	 * @return void
 	 */
-	public static function remove_query_string( $url ) : string {
-		return preg_replace( '/\?.*/', '', $url );
+	public static function write_log( $log ): void {
+		// phpcs:disable
+		if ( is_array( $log ) || is_object( $log ) ) {
+			error_log( print_r( $log, true ) );
+		} else {
+			error_log( $log );
+		}
+		// phpcs:enable
 	}
 
 	/**
-	 * Sanitize a string removing any leading or trailing slashes..
+	 * Retrieves the attachment ID from the file URL.
+	 *
+	 * @param string $image_url The image URL.
 	 *
-	 * @param string $str The string to sanitize.
+	 * @link https://pippinsplugins.com/retrieve-attachment-id-from-image-url/
 	 */
-	public static function unslash_leading_trailing( $str ) : string {
-		$str = untrailingslashit( $str );
-		$str = ltrim( $str, '/' );
+	public static function get_attachment_id( $image_url ): ?string {
```

Comment:
@josephfusco 
Question here: does the : ?string mean that an optional string is returned, or else the usual void type? Just curious, as I haven't seen this before.

Reply:
@jdashley27 Correct that's a nullable return type so in this case a string or null can be returned.

More info here https://www.php.net/manual/en/migration71.new-features.php